### DISCUSSION TITLE
Auto rescan

### DISCUSSION THREAD

**Discussion by kvalev** [photoprism/photoprism#889] [Q&A] [answered]
I am trying to understand what's the best way to automatically rescan my photo library, so I hope that you can point me to the right direction.
I have my photos in a public cloud, but have a local mirror, which is continuously synced. I have mounted the the local mirror as the "originals" folder in photoprism (as read-only, as I dont want to change the mirrored data), but it seems that the data is not being reindexed automatically. I thought about using the imports folder instead, but I dont want to create multiple copies of my data (as I have few hundred gigabytes worth of pictures and videos), and I dont want to modify my mirrored data in any way. Plus my pictures are already sorted in folders.
I looked into the existing auto-indexing option, but it seems this one is only applicable for WebDAV or the imports folder.
So far the only idea that I have is to either run the CLI tool to reindex or periodically call the index REST API (which will be easier in my docker-compose setup).
Another caveat in my setup is that the index operation might run while files are being synced, so the operation might try to reindex (still) incomplete files, but it seems that a complete rescan solves those issues.

What do you think is the best way to trigger the auto rescan in my case?

**Comment by lastzero** [comment, marked as answer]
Syncing via WebDAV is not an option? Continuous changes will always be an issue as files and folders should be "stable" when indexing.

**Reply by kvalev** [reply]
I am not that familiar with WebDAV, but lets say I start a local container to serve my mirrored images - wouldnt PhotoPrism just sync all the data from WebDAV and thus creating another copy?

**Reply by lastzero** [reply]
Yes, syncing two times doesn't make sense and would create two copies. If you want to keep everything as it is, it's probably best to run the CLI command (-a is not necessary) at times you don't expect a lot of changes.

**Comment by tootbrute** [comment]
> What do you think is the best way to trigger the auto rescan in my case?

Is a public cloud, like a VPS? If it is, you could make a cron job to run every night at midnight.

59 23  * * * root docker-compose exec photoprism photoprism index
